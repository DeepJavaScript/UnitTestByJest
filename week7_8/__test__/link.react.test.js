"use strict";

import React from "react";
import Link from "../Link.react";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(<Link page="http://www.facebook.com">Facebook</Link>)
    .toJSON();
  expect(tree).toMatchInlineSnapshot(`
    <a
      className="normal"
      href="http://www.facebook.com"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      Facebook
    </a>
  `);
});

// 把 facebook 改成 ig
// it('renders correctly', () => {
//   const tree = renderer
//     .create(<Link page="http://www.instagram.com">Instagram</Link>)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });

// 總是失敗
it.skip('will fail every time', () => {
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };

  expect(user).toMatchSnapshot();
});

it('will check the matchers and pass', () => {
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };

  expect(user).toMatchSnapshot({
    createdAt: expect.any(Date),
    id: expect.any(Number),
  });
});

it('will check the values and pass', () => {
  const user = {
    createdAt: new Date(),
    name: 'Bond... James Bond',
  };

  expect(user).toMatchSnapshot({
    createdAt: expect.any(Date),
    name: 'Bond... James Bond',
  });
});