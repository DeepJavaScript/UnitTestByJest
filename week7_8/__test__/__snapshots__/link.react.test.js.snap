// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`will check the matchers and pass 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "LeBron James",
}
`;

exports[`will check the values and pass 1`] = `
Object {
  "createdAt": Any<Date>,
  "name": "Bond... James Bond",
}
`;

exports[`will fail every time 1`] = `
Object {
  "createdAt": 2020-05-04T07:52:08.062Z,
  "id": 13,
  "name": "LeBron James",
}
`;
